imap/imap.c

```diff
@@ -165,7 +165,7 @@ static void set_flag(struct ImapAccountData *adata, int aclbit, int flag,
                      const char *str, char *flags, size_t flsize)
 {
   if (mutt_bit_isset(adata->mailbox->rights, aclbit))
-    if (flag && imap_has_flag(&adata->flags, str))
+    if (flag && imap_has_flag(&((struct ImapMboxData *)adata->mailbox->mdata)->flags, str))
```

Comment:
This lengthy casting looks a bit nasty. Using `imap_mdata_get(adata->mailbox)` would improve readability.